src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/ApiBehaviorConventions/ApiVisibilityConvention.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationModels
+{
+    /// <summary>
+    /// A <see cref="IActionModelConvention"/> that sets Api Explorer visibility.
+    /// </summary>
+    public class ApiVisibilityConvention : IActionModelConvention
+    {
+        public void Apply(ActionModel action)
+        {
+            if (!ShouldApply(action))
+            {
+                return;
+            }
+
+            if (action.ApiExplorer.IsVisible == null)
+            {
+                // Enable ApiExplorer for the controller if it wasn't already explicitly configured.
+                action.ApiExplorer.IsVisible = true;
+            }
+        }
```

Comment:
Is this right? This would overwrite visibility if it's turned off at the controller/application level.

Reply:
We [previously ignored](https://github.com/aspnet/Mvc/blob/release/2.2/src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs#L69-L74) the application visibility, but I guess I could check the controller visibility here.